com.dynamo.cr/com.dynamo.cr.bob/src/com/defold/extension/pipeline/ILuaTranspiler.java

```diff
@@ -0,0 +1,114 @@
+// Copyright 2020-2024 The Defold Foundation
+// Copyright 2014-2020 King
+// Copyright 2009-2014 Ragnar Svensson, Christian Murray
+// Licensed under the Defold License version 1.0 (the "License"); you may not use
+// this file except in compliance with the License.
+//
+// You may obtain a copy of the License, together with FAQs at
+// https://www.defold.com/license
+//
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+package com.defold.extension.pipeline;
+
+import java.io.File;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Implement this interface in com.defold.extension.pipeline package with no-argument constructor
+ * to add support for transpiling a language to Lua that is then will be used during the build
+ */
+public interface ILuaTranspiler {
+
+    /**
+     * @return A resource path to a build file, a project-relative path starting with slash,
+     * e.g. {@code "/tlconfig.lua"}
+     */
+    String getBuildFileResourcePath();
+
+    /**
+     * @return A file extension for the source code files of the transpiled language,
+     * e.g. {@code "ext"}
+     */
+    String getSourceExt();
+
+    /**
+     * Build the project from the source dir to output dir
+     *
+     * @param pluginDir a dir inside the project root that contains unpacked plugins that may be
+     *                  used for compilation (i.e. transpiler binaries)
```

Comment:
I think it requires too much backtracking to figure out what is a plugin root for a particular dependency, and it is trivial to figure out from the dependency itself — because it knows its root library directory, so I don't think it's worth it.

Reply:
Even if we don't want to change it, the comment could still do a better job of explaining what this directory is.

Reply:
Okay, I'll add the comment!